core/templates/components/forms/custom-forms-directives/image-uploader.directive.ts

```diff
@@ -51,18 +53,54 @@ angular.module('oppia').directive('imageUploader', [
             return 'This file is not recognized as an image.';
           }
 
-          if (!file.type.match('image/jpeg') &&
-            !file.type.match('image/gif') &&
-            !file.type.match('image/jpg') &&
-            !file.type.match('image/png') &&
-            !file.type.match('image/svg\\+xml')) {
+          var imageTypeMapping = {
+            jpeg: {
+              format: 'image/jpeg',
+              fileType: /jp(e?)g$/,
+              fileExtension: /\.jp(e?)g$/,
+            },
+            jpg: {
+              format: 'image/jpg',
+              fileType: /jp(e?)g$/,
+              fileExtension: /\.jp(e?)g$/,
+            },
+            gif: {
+              format: 'image/gif',
+              fileType: /gif$/,
+              fileExtension: /\.gif$/,
+            },
+            png: {
+              format: 'image/png',
+              fileType: /png$/,
+              fileExtension: /\.png$/,
+            },
+            svg: {
+              format: 'image/svg\\+xml',
+              fileType: /svg\+xml$/,
+              fileExtension: /\.svg$/,
+            }
+          };
+          var imageHasInvalidFormat = true;
+          var imageHasInvalidExtension = false;
+          for (var i = 0; i < scope.getAllowedImageFormats().length; i++) {
+            var fileType = scope.getAllowedImageFormats()[i];
+            if (!(fileType in imageTypeMapping)) {
+              return 'Unknown image format in allowed image formats.';
+            }
+            imageHasInvalidFormat = (
+              imageHasInvalidFormat &&
+              !file.type.match(imageTypeMapping[fileType].format));
+            imageHasInvalidExtension = (
+              imageHasInvalidExtension ||
+              (file.type.match(imageTypeMapping[fileType].fileType) &&
```

Comment:
Confusing to have fileType mean two different things -- please pick a different name for the dummy var.

Reply:
Done.